src/interface_py/h2o4gpu/util/gpu.py

```diff
@@ -54,15 +56,20 @@ def get_gpu_info(return_usage=False):
     res = None
     # sometimes hit broken process pool in cpu mode,
     # so just return back no gpus.
-    try:
-        with ProcessPoolExecutor(max_workers=1) as executor:
-            future = executor.submit(get_gpu_info_subprocess, return_usage)
-            res = future.result()
-        return res
-    except concurrent.futures.process.BrokenProcessPool:
-        if return_usage:
-            return (total_gpus, total_mem, gpu_type, usage)
-        return (total_gpus, total_mem, gpu_type)
+    for trial in range(0, trials):
+        try:
+            with ProcessPoolExecutor(max_workers=1) as executor:
```

Comment:
I was wondering for some time - why are we doing this with an executor? Can't we just call `get_gpu_info`? Since we are doing `future.result` anyway which I assume is synchronous? Wouldn't that fix the issue?

Ah and a minor one - pylint is complaining.

Reply:
I mention why.  py3nvml can create a cuda context and one can't share those across into a child process.  So we have to isolate any cuda/nvml calls out of the parent.

Reply:
Uh yeah my bad for not reading the docstring, sorry.